src/CoreRCON/RCON.cs

```diff
@@ -166,6 +183,11 @@ await writer.FlushAsync()
                 await writer.CompleteAsync()
                     .ConfigureAwait(false);
                 _connected = false;
+                if (_pipeCts != null && !_pipeCts.IsCancellationRequested)
+                {
+                    _pipeCts.Cancel(); // Tell reader to stop waiting
+                    _pipeCts = null;
+                }
```

Comment:
This should not be needed. 
Did you run into cases where the reader waited after the writer was completed. 
There might be need to run `FlushAsync` here instead.

Reply:
You are correct this code has no function, removing it